pkg/compactor/compactor.go

```diff
@@ -588,6 +591,19 @@ func (c *Compactor) starting(ctx context.Context) error {
 		return errors.Wrap(err, "failed to start the blocks cleaner")
 	}
 
+	if c.compactorCfg.CachingBucketEnabled {
+		// Turn off chunksCache since compactor is using get to download chunks, while store-gateway and querier is using getrange.
+		c.storageCfg.BucketStore.ChunksCache.Backend = ""
```

Comment:
Let's not modify the existing config? I wonder if it can cause some problems when Cortex is running multiple targets, both compactor and store gateway.
Let's just pass an empty struct instead.

```
cortex_tsdb.CreateCachingBucket(cortex_tsdb.ChunksCacheConfig{}, ...)
```